### ISSUE TITLE
Array and string offset access syntax with curly braces is deprecated

### ISSUE THREAD

**Issue by timbertens** [clearbold/craft-campaignmonitor-lists#10]
On PHP 7.4.x curly braces are no longer supported, resulting in an error when using this component. 
The problem actually occurs in one of the dependencies, being campaignmonitor/createsend-php
The issue over there is solved since version 6.1.1 of that component, see https://github.com/campaignmonitor/createsend-php/issues/101

But this component still uses version 5.1.3 of the mentioned dependency.  Could you please update the composer.json? 
This is related to https://github.com/clearbold/craft-campaignmonitor-service/issues/3 (and might be to all your components using campaignmonitor/createsend-php)

**Comment by bartrylant** [comment]
Any updates on this issue? Plugin is unusable like this…

**Comment by heymarkreeves** [comment]
Sorry about that. I've pushed an update with the updated requirement. Let me know how that works?

**Comment by timbertens** [comment]
It breaks everything I'm afraid.  Apparently CM requires now to send a ConsentToTrack field, see https://www.campaignmonitor.com/api/subscribers/
I've trying to add it to ListService.php, but now it fails with 'Failed to deserialize your request.'

**Comment by heymarkreeves** [comment]
I quickly added that to v1.1.11. Does that fix it?

**Comment by timbertens** [comment]
yes, of course it has to be 'yes' not true like I tried ... too many hours behind my screen .. 
It works now !

**Comment by timbertens** [comment]
Can you push a new version to make sure ?

**Comment by heymarkreeves** [comment]
Great! v1.1.11 is the new version. You should see that as an available update via Composer.

Do you think consent to track should be a configurable setting or parameter rather than defaulted to 'yes'?

**Comment by timbertens** [comment]
yes, I just noticed it was already a new version ...  great & thanks
Yeah maybe in an ideal world this should be configurable or to be included in the form, but for now that's ok !
